### PR TITLE
Use a ready event handler

### PR DESCRIPTION
**Pull request by xkjyeah**
This creates an event handler ready which is called after the control is populated

Example:
```
<ui-gmap-google-map
    control="control"
    ready="onmapready()"
		>
</ui-gmap-google-map>
```
Thus you can write:
```javascript
// Controller function
function($scope, uiGmapGoogleMapApi) {
    $scope.control = {}
    $scope.onmapready = function() {
        console.log($scope.control.getGMap) // not undefined
        $scope.control.getGMap().fitBounds(myBounds);
    }
}
```

This obviates the headache with tracking instance numbers using `uiGmapIsReady`

### CODE REVIEW

**Comment by cthrax** [comment]
Looks like this deletes all the build artifacts, since that's part of our distribution, that's a bad idea.

**Comment by xkjyeah** [comment]
Ok fixed that, but Travis has some problem install NPM :/

File: src/coffee/directives/api/map.coffee [hunk 1]
```diff
@@ -115,6 +116,11 @@ angular.module('uiGmapgoogle-maps.directives.api')
               scope.deferred.resolve _gMap
               resolveSpawned()
 
+            # run the ready event
+            if angular.isDefined scope.ready
+              scope.$applyAsync ->
```
Comment (nmccready) [inline src/coffee/directives/api/map.coffee]: This is at the wrong indention level. This needs to be inside the idle callback, below `resolveSpawned()`.

Also the logic seems terse. Just do scope.`$parent.$evalAsync scope.ready if angular.isDefined scope.ready `

**Comment by nmccready** [comment]
Closing since this is a 1 month stale.

**Comment by nmccready** [comment]
Will re-open if the discussion is continued.

**Pull request closed without merging**